integration/yarn/src/main/java/alluxio/yarn/ApplicationMaster.java

```diff
@@ -312,13 +317,21 @@ public void start() throws IOException, YarnException {
    * @throws Exception if an error occurs while requesting or launching containers
    */
   public void requestAndLaunchContainers() throws Exception {
-    Resource masterResource = Records.newRecord(Resource.class);
-    masterResource.setMemory(mMasterMemInMB);
-    masterResource.setVirtualCores(mMasterCpu);
-    mContainerAllocator = new ContainerAllocator("master", 1, 1, masterResource, mYarnClient,
-        mRMClient, mMasterAddress);
-    List<Container> masterContainers = mContainerAllocator.allocateContainers();
-    launchMasterContainer(Iterables.getOnlyElement(masterContainers));
+
+    if (masterExists()) {
+      InetAddress address = InetAddress.getByName(mMasterAddress);
+      mMasterContainerNetAddress = address.getHostAddress();
+      LOG.info("Found master already running on " + mMasterAddress);
+    } else {
+      LOG.info("Configuring Master container.");
```

Comment:
maybe change to "Configuring master container request" since the container hasn't been allocated yet